src/EnergyPlus/api/datatransfer.py

```diff
@@ -950,3 +1006,311 @@ def kind_of_sim(self, state: c_void_p) -> int:
         :return: Integer value of current environment.
         """
         return self.api.kindOfSim(state)
+
+    def today_weather_is_raining_at_time(self, state: c_void_p, hour: int, time_step_number: int) -> bool:
```

Comment:
Add Python wrappings around each API function to allow a client to call into this from either Python EMS or through raw API calls during a callback in a running simulation.